azure/scope/managedcontrolplane.go

```diff
@@ -265,6 +268,18 @@ func (s *ManagedControlPlaneScope) SetSubnet(_ infrav1.SubnetSpec) {
 	// no-op
 }
 
+// UpdateSubnetCIDRs updates the subnet CIDRs for the subnet with the same name.
+// This is not used when using a managed control plane.
+func (s *ManagedControlPlaneScope) UpdateSubnetCIDRs(_ string, _ []string) {
```

Comment:
You could write a unit test for this function that tested that it didn't mutate `s` (or parsed its AST to see that its body was actually a no-op). But that's probably not worth the effort. 😄

Reply:
But let's not. :-p